.github/workflows/full_kubeflow_integration_test.yaml

```diff
@@ -4,278 +4,178 @@ on:
-      uses: actions/setup-python@v4
-      with:
-        python-version: '3.12'
-    - name: Install Python Dependencies
+
+    - name: Install Jupyter and Notebook Components
       run: |
+        kustomize build apps/jupyter/jupyter-web-app/upstream/overlays/istio/ | kubectl apply -f -
+        kustomize build apps/jupyter/notebook-controller/upstream/overlays/kubeflow/ | kubectl apply -f -
+        kustomize build apps/admission-webhook/upstream/overlays/cert-manager | kubectl apply -f -
+        kubectl get crd poddefaults.kubeflow.org || kubectl apply -f https://raw.githubusercontent.com/kubeflow/kubeflow/master/components/admission-webhook/manifests/base/crd.yaml
+        kubectl wait --for=condition=Ready pods --all --all-namespaces --timeout 300s --field-selector=status.phase!=Succeeded
+
+    - name: Install ML Components
+      run: |
+        # Katib
+        sudo apt-get install -y apparmor-profiles
+        sudo apparmor_parser -R /etc/apparmor.d/usr.sbin.mysqld
+        cd apps/katib/upstream && kustomize build installs/katib-with-kubeflow | kubectl apply -f - && cd ../../../
+        kubectl wait --for=condition=Available deployment/katib-controller -n kubeflow --timeout=300s
+        kubectl wait --for=condition=Available deployment/katib-mysql -n kubeflow --timeout=300s
+        kubectl label namespace $KF_PROFILE katib.kubeflow.org/metrics-collector-injection=enabled --overwrite
+
+        # Other ML components
+        ./tests/gh-actions/install_training_operator.sh
+        ./tests/gh-actions/install_knative.sh
+        ./tests/gh-actions/install_kserve.sh
+        chmod u+x tests/gh-actions/spark_*.sh && ./tests/gh-actions/spark_install.sh
+
+        # Dependencies
         pip install pytest kubernetes kfp==2.11.0 kserve pytest-timeout pyyaml requests
-    - name: port forward
-      run: |
-        ingress_gateway_service=$(kubectl get svc --namespace istio-system --selector="app=istio-ingressgateway" --output jsonpath='{.items[0].metadata.name}')
-        nohup kubectl port-forward --namespace istio-system svc/${ingress_gateway_service} 8080:80 &
-        while ! curl localhost:8080; do echo waiting for port-forwarding; sleep 1; done; echo port-forwarding ready
-    - name: test dex login
+
+    - name: Setup Gateway
       run: |
-        pip3 install requests
-        ./tests/gh-actions/test_dex_login.py
-    - name: Run ML Pipeline Integration Tests
+        GATEWAY=$(kubectl get svc -n istio-system -l app=istio-ingressgateway -o jsonpath='{.items[0].metadata.name}')
+        nohup kubectl port-forward -n istio-system svc/$GATEWAY 8080:80 &
+        while ! curl -s localhost:8080 > /dev/null; do sleep 1; done
+
+    - name: Test Dex Authentication
       run: |
-        KF_PROFILE=kubeflow-user-example-com
+        # Authentication
+        chmod +x tests/gh-actions/test_dex_auth.sh && ./tests/gh-actions/test_dex_auth.sh
 
-        # Test with authorized token (authorized user flow)
+        # ML Pipeline
```

Comment:
```suggestion
```

Reply:
One step per component. Pipeline authentication belongs to the pipeline section.

Reply:
Where is the pipeline test script? there is way too much code here.